api/src/main/java/jakarta/security/enterprise/authentication/mechanism/http/BasicAuthenticationMechanismDefinition.java

```diff
@@ -16,30 +17,75 @@
 
 package jakarta.security.enterprise.authentication.mechanism.http;
 
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
 import static java.lang.annotation.ElementType.TYPE;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
+import jakarta.enterprise.util.AnnotationLiteral;
+import jakarta.inject.Qualifier;
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
 
 /**
  * Annotation used to define a container authentication mechanism that implements
- * the HTTP basic access authentication protocol as defined by the Servlet spec (13.6.1)  
+ * the HTTP basic access authentication protocol as defined by the Servlet spec (13.6.1)
  * and make that implementation available as an enabled CDI bean.
  *
  */
 @Retention(RUNTIME)
 @Target(TYPE)
 public @interface BasicAuthenticationMechanismDefinition {
-    
+
     /**
      * Name of realm that will be sent via the <code>WWW-Authenticate</code> header.
      * <p>
      * Note that this realm name <b>does not</b> couple a named identity store
      * configuration to the authentication mechanism.
-     * 
+     *
      * @return Name of realm
      */
     String realmName() default "";
+
+    /**
+     * List of {@link Qualifier qualifier annotations}.
+     *
+     * <p>
+     * An {@link HttpAuthenticationMechanism} injection point
+     * with these qualifier annotations injects a bean that is
+     * produced by this {@code BasicAuthenticationMechanismDefinition}.</p>
+     *
+     * <p>
+     * The default value is {@code BasicAuthenticationMechanism}, indicating that
+     * this {@code BasicAuthenticationMechanismDefinition} produces
+     * bean instances of type {@link HttpAuthenticationMechanism} qualified by
+     * {@code BasicAuthenticationMechanism}.
```

Comment:
The current behavior is that the produced bean has no qualifier and can be injected with `@Inject HttpAuthenticationMechanism`. 

Shouldn't we keep this behavior? I mean, in case the default BasicAuthenticationMechanism qualifier is used here, shouldn't we create a HAM with the no qualifiers instead of with the `BasicAuthenticationMechanism` qualifier? It implies that if there are more definitions, only one can use the default qualifier. And then we don't need a separate qualifier for each authentication type, only one, e.g. `DefaultAuthenticationMechanism` instead of `BasicAuthenticationMechanism` and similar.